tests/integration/server/meeting/meeting.controller.test.ts

```diff
@@ -0,0 +1,1239 @@
+                        ...mtg, roomId: room?.id,
+                      })),
+                    });
+                  result = response.body;
+                  savedMeeting = result
+                    .find(({ id }) => id === meetingToEdit.id);
+                });
+                it('Should update the meeting day', function () {
+                  strictEqual(savedMeeting.day, updatedMeeting.day);
+                });
+                it('Should update the meeting startTime', function () {
+                  const updatedStartTime = tzStringToAMPM(
+                    updatedMeeting.startTime
+                  );
+                  strictEqual(savedMeeting.startTime, updatedStartTime);
+                });
+                it('Should update the meeting endTime', function () {
+                  const updatedEndTime = tzStringToAMPM(
+                    updatedMeeting.endTime
+                  );
+                  strictEqual(savedMeeting.endTime, updatedEndTime);
+                });
+                it('Should change the room', function () {
+                  strictEqual(savedMeeting.room.id, unbookedRoom.id);
+                  notStrictEqual(savedMeeting.room.id, meetingToEdit.room.id);
+                  notStrictEqual(savedMeeting.room, undefined);
+                });
+              });
+              context('When the room is already booked', function () {
+                let bookedRoom: Room;
+                beforeEach(async function () {
+                  ([
+                    meetingToEdit,
+                    ...otherMeetings
+                  ] = testNonClassEvent.meetings);
+                  updatedMeeting = new Meeting();
+                  Object.assign(updatedMeeting, meetingToEdit);
+                  updatedMeeting.day = DAY.MON;
+                  updatedMeeting.startTime = '10:00:00-05';
+                  updatedMeeting.endTime = '16:00:00-05';
+                  const bookedRoomQuery = roomRepository
+                    .createQueryBuilder('r')
+                    .where('r.id <> :roomId', { roomId: updatedMeeting.room.id })
+                    .andWhere('"meetingTitle" IS NOT NULL')
+                    .leftJoin(RoomBookingInfoView, 'rb', `rb.roomId = r.id AND rb.day = '${DAY.MON}'`);
+                  bookedRoom = await bookedRoomQuery
+                    .getOne();
+                  updatedMeeting.room = bookedRoom;
+                  response = await request(api)
+                    .put(`/api/meetings/${testNonClassEvent.id}`)
+                    .send({
+                      meetings: [
+                        updatedMeeting,
+                        ...otherMeetings,
+                      ].map(({ room, ...mtg }) => ({
+                        ...mtg, roomId: room?.id,
+                      })),
+                    });
+                });
+                it('Should return a Bad Request error', function () {
+                  strictEqual(response.status, 400);
```

Comment:
I'd suggest using the constants here for readability, i.e: `status.BAD_REQUEST`. Though, I'm not set on it - I'll leave that up to you.